docs/guide/nested-module-subdirs.md

```diff
@@ -0,0 +1,113 @@
+# Nested module folders
+
+Apostrophe has optional support for nested subdirectories of modules, tucked
```

Comment:
This introductory paragraph could be clearer about what the feature is. I'd suggest leading with a problem this solves to contextualize it. "As website projects grow, code organization is increasingly important. To help with this, Apostrophe support optionally nesting module directories and registering them as module clusters."

I threw in the clustering bit. Simply nesting is a codebase file benefit. Registering them as groups with a `modules.js` file in app.js is an additional code organization benefit, considering these modules as a group (or cluster) of some sort.